crypto/stack/stack.c

```diff
@@ -322,17 +327,39 @@ static int internal_find(OPENSSL_STACK *st, const void *data,
     r = ossl_bsearch(&data, st->data, st->num, sizeof(void *), st->comp,
                      ret_val_options);
 
+    if (pnum != NULL) {
+        if (r != NULL) {
+            const void **p = (const void **)r;
```

Comment:
Why the cast?

Reply:
It is necessary to cast away the const because here it is at different indirection level.